src/BizHawk.Emulation.Cores/Computers/Doom/DSDA.ISettable.cs

```diff
@@ -0,0 +1,316 @@
+﻿using System.ComponentModel;
+
+using BizHawk.Emulation.Common;
+using BizHawk.Common;
+using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
+
+namespace BizHawk.Emulation.Cores.Computers.Doom
+{
+	public partial class DSDA : ISettable<DSDA.DoomSettings, DSDA.DoomSyncSettings>
+	{
+		public enum CompatibilityLevelEnum : int
+		{
+			[Display(Name = "0 - Doom v1.2")]
+			C0 = 0,
+			[Display(Name = "1 - Doom v1.666")]
+			C1 = 1,
+			[Display(Name = "2 - Doom v1.9")]
+			C2 = 2,
+			[Display(Name = "3 - Ultimate Doom & Doom95")]
+			C3 = 3,
+			[Display(Name = "4 - Final Doom")]
+			C4 = 4,
+			[Display(Name = "5 - DOSDoom")]
+			C5 = 5,
+			[Display(Name = "6 - TASDoom")]
+			C6 = 6,
+			[Display(Name = "7 - Boom's Inaccurate Vanilla Compatibility Mode")]
+			C7 = 7,
+			[Display(Name = "8 - Boom v2.01")]
+			C8 = 8,
+			[Display(Name = "9 - Boom v2.02")]
+			C9 = 9,
+			[Display(Name = "10 - LxDoom")]
+			C10 = 10,
+			[Display(Name = "11 - MBF")]
+			C11 = 11,
+			[Display(Name = "12 - PrBoom v2.03beta")]
+			C12 = 12,
+			[Display(Name = "13 - PrBoom v2.1.0")]
+			C13 = 13,
+			[Display(Name = "14 - PrBoom v2.1.1 - 2.2.6")]
+			C14 = 14,
+			[Display(Name = "15 - PrBoom v2.3.x")]
+			C15 = 15,
+			[Display(Name = "16 - PrBoom v2.4.0")]
+			C16 = 16,
+			[Display(Name = "17 - PrBoom Latest Default")]
+			C17 = 17,
+			[Display(Name = "21 - MBF21")]
+			C21 = 21
+		}
+
+		public enum SkillLevelEnum : int
+		{
+			[Display(Name = "1 - I'm too young to die")]
+			S1 = 1,

```

Comment:
If `.lmp` serialises these as 0 though 4, we should too.

Reply:
1-5 is the [canonical numeration](https://doom.fandom.com/wiki/Skill_level#Doom,_Doom_II_and_Final_Doom_skill_levels). This is also how you pass the desired skill level to DSDA and Vanilla Doom as argument (-skill 1 -> easiest). It's only in the lmp that this value is normalized to zero, but this would be cause for confusion for anybody else using the interface in a normal way.